server/server.js

```diff
@@ -114,8 +124,20 @@ const startServer = (htmlPages) => {
         express.static(path.resolve(rootDirectory, 'images'))
     );
 
+    // Give users fallback locations from local file if aduser is unresponsive
```

Comment:
Synes muligens hardkodet fallback-data er litt overkill her, da har vi enda et sted vi må vedlikeholde/huske å oppdatere data for. Hvis aduser er helt nede (ansett som en av to primære backends for pam-stilligsok) over lengre tid er ting fscked generelt, og det er ikke en normal-situasjon som man skulle trenge dette for. Det gjør også koden mer komplisert. Kan man ikke, så lenge det feiler fra aduser, bare forsøke på nytt og nytt for hver innkommende request ? Inntil man får data fra aduser vil det altså feile ut, og det synes hvertfall jeg er en helt grei avgrensning på hvor langt man tar den.

Reply:
Er ikke ofte fylker/kommuner forandrer seg, vedlikeholdet er minimalt. Men er muligens litt overkill ja, selv om det er fint å være helt safe. Er ikke spesiell komplisert kode heller. Drar bare et json objekt ut av en fil.

Reply:
Jeg forstår ditt synspunkt, og vet det ikke er veldig komplisert. Men alt av slike nice to have ekstra-features akkumulerer seg opp i alle appene og koden vi administrerer og vedlikeholder, til en total som er meget stor (jeg har vært med på KOR2020 i Team Arbeidsplassen 😄 ). Det enkleste her hadde vært å ikke hardkode noenting, og bare la den være så dum som mulig ved å alltid hente data fra kilden, samt benytte cache-headere. Da fører det ikke til noe mer vedlikehold der. Gevinsten med denne fallback-løsningen (helt safe) er lavere enn den vedlikeholdskostnaden som det legger til for teamet, synes jeg.

Reply:
Jeg kan foreslå følgende: mellomlagre første og beste vellykkede geo-response fra aduser i minnet i nodejs, on-demand, ved første kall fra browser. Deretter for påfølgende kall: finnes det noe cachet (en verdi i et geo-json-objekt i en eller annen kontekst i node-server), så serves det rett ut (og alltid med `Cache-control: max-age=86400`-ish). Super-enkel pull-through cache, trenger ikke expiry en gang. Da slipper vi hardkoding av data og vedlikehold.

(Det kjedelige vedlikeholdet består i at man hver gang må lage den statiske filen på nytt når pam-geography oppdateres med nye postnumre, fordi det kan være navne-endringer på kommuner, f.eks. korreksjon av skrivemåter, etc.)

Reply:
Har diskutert litt med Ola også nå. Fjerner hardkoded fil, men beholder in memory cache.